In [1]:
import pandas as pd
from pathlib import Path
import numpy as np
import pandas as pd



In [6]:
training_size_to_score_bullfrog = {}
bullfrog_score =  [0.9511111111111111, 0.9480888888888889, 0.9415111111111111, 0.9411555555555556, 0.9715555555555556, 0.9511111111111111, 0.9612444444444445, 0.9678222222222223, 0.9480888888888889, 0.9749333333333332, 0.9616,0.948088888888889]

j = 0
for i in range(25, 301, 25):
    training_size_to_score_bullfrog[i] = bullfrog_score[j]
    j += 1

training_size_to_score_bullfrog

{25: 0.9511111111111111,
 50: 0.9480888888888889,
 75: 0.9415111111111111,
 100: 0.9411555555555556,
 125: 0.9715555555555556,
 150: 0.9511111111111111,
 175: 0.9612444444444445,
 200: 0.9678222222222223,
 225: 0.9480888888888889,
 250: 0.9749333333333332,
 275: 0.9616,
 300: 0.948088888888889}

In [7]:
training_size_to_score_coyote = {}
coyote_score = [0.9468000000000001, 0.9808, 0.9716, 0.9724, 0.9904, 0.9803999999999999, 0.9787999999999999, 0.9768]

j = 0
for i in range(25, 201, 25):
    training_size_to_score_coyote[i] = coyote_score[j]
    j += 1

training_size_to_score_coyote

{25: 0.9468000000000001,
 50: 0.9808,
 75: 0.9716,
 100: 0.9724,
 125: 0.9904,
 150: 0.9803999999999999,
 175: 0.9787999999999999,
 200: 0.9768}

In [9]:
training_size_to_score_engine = {}
engine_score = [0.8400000000000001, 0.8479999999999999, 0.824, 0.8896]
j = 0
for i in range(25, 101, 25):
    training_size_to_score_engine[i] = engine_score[j]
    j += 1

training_size_to_score_engine

{25: 0.8400000000000001, 50: 0.8479999999999999, 75: 0.824, 100: 0.8896}

In [10]:
training_size_to_score_human_vocal = {}
human_vocal = [0.8622222222222222, 0.8933333333333333, 0.8977777777777778]
training_size_to_score_human_vocal[25] = human_vocal[0]
training_size_to_score_human_vocal[50] = human_vocal[1]
training_size_to_score_human_vocal[60] = human_vocal[2]
training_size_to_score_human_vocal

{25: 0.8622222222222222, 50: 0.8933333333333333, 60: 0.8977777777777778}

In [13]:
training_size_to_score_WT = {}
woodhouses_toad_score = [0.9921777777777778, 0.9994666666666667, 1.0, 1.0, 1.0, 0.9975111111111111, 1.0, 0.9811555555555556, 1.0, 1.0,1.0, 0.9991111111111112]
len(woodhouses_toad_score)
j = 0
for i in range(25, 301, 25):
    training_size_to_score_WT[i] = woodhouses_toad_score[j]
    j += 1

training_size_to_score_WT

{25: 0.9921777777777778,
 50: 0.9994666666666667,
 75: 1.0,
 100: 1.0,
 125: 1.0,
 150: 0.9975111111111111,
 175: 1.0,
 200: 0.9811555555555556,
 225: 1.0,
 250: 1.0,
 275: 1.0,
 300: 0.9991111111111112}

In [15]:
training_size_to_score_pcf = {}
pcf_score = [0.5640888888888889, 0.5662222222222222, 0.6042666666666666, 0.5447111111111111, 0.5992888888888889, 0.5685333333333333, 0.5532444444444444, 0.5564444444444444, 0.5504 ,0.5534222222222223 ,0.5223111111111112, 0.5221333333333333]
j = 0
for i in range(25, 301, 25):
    training_size_to_score_pcf[i] = pcf_score[j]
    j += 1

training_size_to_score_pcf

{25: 0.5640888888888889,
 50: 0.5662222222222222,
 75: 0.6042666666666666,
 100: 0.5447111111111111,
 125: 0.5992888888888889,
 150: 0.5685333333333333,
 175: 0.5532444444444444,
 200: 0.5564444444444444,
 225: 0.5504,
 250: 0.5534222222222223,
 275: 0.5223111111111112,
 300: 0.5221333333333333}

In [16]:
training_size_to_score_field_cricket = {}
field_cricket_score= [0.850864, 0.796543, 0.807407, 0.815802, 0.777778, 0.833580]
j = 0
for i in range(25, 151, 25):
    training_size_to_score_field_cricket[i] = field_cricket_score[j]
    j += 1

training_size_to_score_field_cricket

{25: 0.850864,
 50: 0.796543,
 75: 0.807407,
 100: 0.815802,
 125: 0.777778,
 150: 0.83358}

In [20]:
data = {"coyote": training_size_to_score_coyote,
        "bullfrog":training_size_to_score_bullfrog,
        "pacific chorus frog": training_size_to_score_pcf,
        "woodhouses toad" : training_size_to_score_WT,
        "field_cricket" : training_size_to_score_field_cricket,
        "human vocal" : training_size_to_score_human_vocal,
        "engine" : training_size_to_score_engine}
df = pd.DataFrame(data)
csv_path = '/Users/yifeitao/Desktop/SoundHub/SoundHub/train_with_perch/summary_data.csv'
df.to_csv(csv_path, index_label='Training Size')

In [21]:
summary_stats = df.describe()
summary_stats

coyote   bullfrog  pacific chorus frog  woodhouses toad  \
count  8.000000  12.000000            12.000000        12.000000   
mean   0.974750   0.955526             0.558756         0.997452   
std    0.012709   0.011530             0.025036         0.005606   
min    0.946800   0.941156             0.522133         0.981156   
25%    0.972200   0.948089             0.548978         0.998711   
50%    0.977800   0.951111             0.554933         1.000000   
75%    0.980500   0.963156             0.566800         1.000000   
max    0.990400   0.974933             0.604267         1.000000   

       field_cricket  human vocal    engine  
count       6.000000     3.000000  4.000000  
mean        0.813662     0.884444  0.850400  
std         0.026097     0.019373  0.027973  
min         0.777778     0.862222  0.824000  
25%         0.799259     0.877778  0.836000  
50%         0.811605     0.893333  0.844000  
75%         0.829136     0.895556  0.858400  
max         0.850864     0.897778  0.889600

The training results of the Google PERCH model indicate strong performance across multiple categories. The "coyote" category achieved an average accuracy of 97.5% with low variability (std = 1.3%), while "bullfrog" had an average of 95.6% with slightly higher variability (std = 1.2%). The "pacific chorus frog" showed moderate performance, averaging 55.9% accuracy, with variability indicating slight instability (std = 2.5%). "Woodhouse's toad" performed exceptionally, averaging 99.7% accuracy with minimal variability (std = 0.6%). Other categories, such as "field cricket," averaged 81.4% with some fluctuations (std = 2.6%), and "human vocal" scored 88.4% with limited data available. Finally, the "engine" category averaged 85.0%, with moderate variability (std = 2.8%). These results suggest the model is highly accurate for certain categories, while others, such as "pacific chorus frog," may benefit from further refinement.

As the training size increased, most categories demonstrated improved performance, suggesting that the model effectively learned better feature representations with larger datasets. However, the extent of this improvement varied significantly among categories, reflecting differences in the complexity of the underlying features and the sufficiency of the data.

For high-performing categories like Coyote and Woodhouse's Toad, the model reached excellent accuracy levels even with smaller training sizes. For instance, the Coyote category achieved 94.7% accuracy with a training size of 25 and stabilized at over 97% as the dataset grew. Similarly, Woodhouse's Toad demonstrated near-perfect performance early on, with 98.1% accuracy at a training size of 25 and 100% accuracy by size 75. These results suggest that the model quickly identified and leveraged the distinct features of these categories, making additional data unnecessary for further improvement. This early stabilization highlights the efficiency of the model in learning simpler or more distinguishable features.

On the other hand, categories like Pacific Chorus Frog and Field Cricket presented challenges, even with larger training sizes. The Pacific Chorus Frog category started at 52.2% accuracy with 25 training samples and improved to 60.4% by size 75, but its performance dipped slightly at higher training sizes, suggesting the need for better data representation or additional feature engineering. Field Cricket showed moderate accuracy improvements, ranging from 77.8% to 85.1%, but variability across training sizes indicated instability in the learning process. These findings suggest that these categories either have more complex features or suffer from insufficient or imbalanced data, necessitating targeted interventions to enhance their performance.

Limited observations for the Human Vocal and Engine categories constrained detailed analysis, but a general upward trend was evident for Engine accuracy as training size increased, ranging from 82.4% to 88.9%. Human Vocal averaged 88.4% accuracy, demonstrating good baseline performance, but additional data points are required to assess its full learning potential. These trends underscore the importance of sufficient training data to stabilize and optimize model performance, particularly for underrepresented classes.

In conclusion, while the model performs exceptionally well for categories like Woodhouse's Toad and Bullfrog with minimal training data, categories such as Pacific Chorus Frog and Field Cricket require additional focus to improve accuracy and stability. Strategies such as reallocating training resources, augmenting data, and rebalancing classes could help address performance disparities. Moreover, analyzing feature quality for underperforming categories might uncover specific challenges in their representations, guiding targeted improvements. Understanding these dynamics allows for optimizing training processes and enhancing the overall performance of the model.